packages/openapi3/src/schema-emitter.ts

```diff
@@ -589,13 +600,14 @@ export class OpenAPI3SchemaEmitter extends TypeEmitter<
       return wrapWithObjectBuilder(schemaMembers[0], { mergeUnionWideConstraints: true });
     }
 
+    const isMerge = checkMerge(schemaMembers);
     const schema: OpenAPI3Schema = {
       [ofType]: schemaMembers.map((m) =>
-        wrapWithObjectBuilder(m, { mergeUnionWideConstraints: false }),
+        wrapWithObjectBuilder(m, { mergeUnionWideConstraints: isMerge }),
```

Comment:
isn't that going to create unecessary warpping?

can you add a test like below that also check this union `Foo | string | null`

Reply:
I have check this, no more warpping. also added new case